src/Service/HaalCentraalToStufBGService.php

```diff
@@ -38,21 +54,119 @@ class HaalCentraalToStufBGService
 
 
     /**
-     * @param EntityManagerInterface $entityManager The Entity Manager.
-     * @param LoggerInterface        $pluginLogger  The plugin version of the logger interface.
+     * @param CallService            $callService            The CallService.
+     * @param MappingService         $mappingService         The MappingService.
+     * @param GatewayResourceService $gatewayResourceService The GatewayResourceService.
+     * @param LoggerInterface        $pluginLogger           The plugin version of the logger interface.
      */
     public function __construct(
-        EntityManagerInterface $entityManager,
+        CallService $callService,
+        MappingService $mappingService,
+        GatewayResourceService $gatewayResourceService,
         LoggerInterface $pluginLogger
     ) {
-        $this->entityManager = $entityManager;
-        $this->logger        = $pluginLogger;
-        $this->configuration = [];
-        $this->data          = [];
+        $this->callService            = $callService;
+        $this->mappingService         = $mappingService;
+        $this->gatewayResourceService = $gatewayResourceService;
+        $this->logger                 = $pluginLogger;
+        $this->configuration          = [];
+        $this->data                   = [];
 
     }//end __construct()
 
 
+    /**
+     * Finds source by reference.
+     *
+     * @return Source|null The resulting source.
+     */
+    public function getSource(): ?Source
```

Comment:
Het idee van de $this->gatewayResourceService->getSource functie is dat je dit niet meer hoeft te doen